ztools/iscsi/scan.go

```diff
@@ -0,0 +1,117 @@
+package iscsi
+
+import (
+	"net"
+	"fmt"
+	"regexp"
+)
+
+// FIXME, this is hacky and may not be very resilient
+var iscsiTargetRegex = regexp.MustCompile(`TargetName=([^\x00]+)[^T]*TargetAddress=([A-Za-z0-9\./:]+)`)
+/*var (
+	iscsiTargetNameRegex = regexp.MustCompile(`TargetName=([^\x00]+)`)
+	iscsiTargetAddressRegex = regexp.MustCompile(`TargetAddress=([A-Za-z0-9\./:]+)`)
+)*/
+
+
+func Scan(conn net.Conn, config *ISCSIConfig) (AuthLog, error) {
+	authlog := AuthLog{conn.RemoteAddr().String(), []Target{}, false}
+	// FIXME, there is too much hardcoded stuff going on here, needs more command line flags
+	body := fmt.Sprintf("%x00%x00%x00%x00%x00%x00%x00%x00%x00%x00%x",
+		"InitiatorName=iqn.1994-05.com.redhat:4fd2932635a0",
+		// Apparently login only accepts names that are % 4 == 1 bytes long.
+		// Integer division plus padding to the rescue!
+		"InitiatorAlias=" + fmt.Sprintf("%0" + fmt.Sprint(len(config.LocalLogin) / 4 * 4 + 5)  + "s", config.LocalLogin),
+		"SessionType=Discovery",
+		"HeaderDigest=None",
+		"DataDigest=None",
+		"DefaultTime2Wait=2",
+		"DefaultTime2Retain=0",
+		"IFMarker=No",
+		"OFMarker=No",
+		"ErrorRecoveryLevel=0",
+		"MaxRecvDataSegmentLength=32768",
+	)
+	header := fmt.Sprintf("438700000000%04x00023D00000000000000000000000000000000010000000000000000000000000000000000000000", len(body)/2)
```

Comment:
From a proof-of-concept script which I ported, which in turn mostly reflected stuff observed in wireshark. I agree that much of this should probably be made configurable.